### PR TITLE
[DOCS] Add 'test' extra to setup.py with docs

### PR DESCRIPTION
**Pull request by kenwade4**
Changes proposed in this pull request:
- Add 'test' extra to setup.py with dev-lite and dev-contrib requirements
- Update contributing setup doc with examples of installing extras

### Definition of Done
Please delete options that are not relevant.

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [x] I have made corresponding changes to the documentation

### CODE REVIEW

File: docs/contributing/contributing_setup.md [hunk 1]
```diff
@@ -58,9 +58,39 @@ source ge_dev/bin/activate
 
 pip install --upgrade pip==21.3.1
 
-pip install -r requirements.txt -r requirements-dev-lite.txt -c constraints-dev.txt -e .
+pip install -c constraints-dev.txt -e ".[test]"
 ```
 
+> Note: You may specify other "extras" in the square brackets next to "test" if you separate with a comma (i.e. `-e ".[test,postgresql,trino]"`)
```
Comment (billdirks) [inline docs/contributing/contributing_setup.md:64]: This is a great and thorough addition!

File: setup.py [hunk 2]
```diff
@@ -38,8 +38,10 @@ def get_extras_require():
             results[key] = parsed
 
     lite = results.pop("lite")
+    contrib = results.pop("contrib")
```
Comment (billdirks) [inline setup.py:41]: This will throw if `contrib` is not present but we probably do want to die in that case.

**Review by billdirks — APPROVED** [review]
LGTM!

This is a very clear way to explain this, thank you!

File: setup.py (hunk 2 above)
Comment (kenwade4) [reply setup.py:41]: The requirements-dev-contrib.txt file exists, so that will never raise an exception.

**Pull request merged**